### PR TITLE
appease rubocop overlords

### PR DESCRIPTION
**Pull request by ndushay**
**NOTE:  Changes to openapi.yml require updating openapi.yml for sdr-api and dor-services-app and generating models - see README.**

## Why was this change made? 🤔

because I got sick of seeing warnings for the new cops when generating schemas.

## How was this change tested? 🤨

changes are to rubocop.yml and 3 spec files;  CI build is sufficient

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -311,3 +311,28 @@ Style/RedundantInitialize: # new in 1.27
...
+Style/MagicCommentFormat: # new in 1.35
+  Enabled: true
+Style/MapCompactWithConditionalBlock: # new in 1.30
+  Enabled: true
+RSpec/ChangeByZero: # new in 2.11.0
+  Enabled: true
+RSpec/Capybara/SpecificMatcher: # new in 2.12
+  Enabled: true
+RSpec/Rails/HaveHttpStatus: # new in 2.12
+  Enabled: true
```
Comment (jcoyne) [inline .rubocop.yml:338]: Can we disable this one since we aren't using rspec-rails?

File: .rubocop.yml [hunk 2]
```diff
@@ -311,3 +311,28 @@ Style/RedundantInitialize: # new in 1.27
...
+Style/EnvHome: # new in 1.29
+  Enabled: true
+Style/MagicCommentFormat: # new in 1.35
+  Enabled: true
+Style/MapCompactWithConditionalBlock: # new in 1.30
+  Enabled: true
+RSpec/ChangeByZero: # new in 2.11.0
+  Enabled: true
+RSpec/Capybara/SpecificMatcher: # new in 2.12
+  Enabled: true
```
Comment (jcoyne) [inline .rubocop.yml:336]: Can we disable this one since we don't have Capybara?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: .rubocop.yml (hunk 1 above)
Comment (ndushay) [reply .rubocop.yml:338]: oh duh - yes.

Comment (ndushay) [reply .rubocop.yml]: done.

File: .rubocop.yml (hunk 2 above)
Comment (ndushay) [reply .rubocop.yml]: done

**Review by jcoyne — APPROVED** [review]

**Pull request merged**